stackslib/src/chainstate/stacks/auth.rs

```diff
@@ -908,9 +1205,35 @@ impl TransactionAuth {
         }
     }
 
-    pub fn from_p2wpkh(privk: &StacksPrivateKey) -> Option<TransactionAuth> {
-        match TransactionSpendingCondition::new_singlesig_p2wpkh(StacksPublicKey::from_private(
-            privk,
+    pub fn from_order_independent_p2sh(
+        privks: &[StacksPrivateKey],
+        num_sigs: u16,
+    ) -> Option<TransactionAuth> {
+        let mut pubks = vec![];
+        for privk in privks.iter() {
+            pubks.push(StacksPublicKey::from_private(privk));
+        }
```

Comment:
Probably more efficient to do

```suggestion
        let pubks = privks.iter()
            .map(|privk| StacksPublicKey::from_private(privk))
            .collect();
```